docs/aws/audit/kmsmonitoring/rules/secrets_manager_in_use_remediation.mdx

```diff
@@ -1,6 +1,256 @@
+
+   ```sh
+   aws secretsmanager create-secret --name MySecretName --secret-string "MySecretValue" --kms-key-id "arn:aws:kms:region:account-id:key/key-id"
+   ```
+
+2. **Enable Automatic Rotation for the Secret:**
+   Enable automatic rotation for the secret to ensure it is regularly updated and managed securely.
+
+   ```sh
+   aws secretsmanager rotate-secret --secret-id MySecretName --rotation-lambda-arn arn:aws:lambda:region:account-id:function:MyRotationFunction
+   ```
+
+3. **Grant Necessary Permissions to KMS Key:**
+   Ensure that the KMS key used to encrypt the secret has the necessary permissions. Use the `put-key-policy` command to attach a key policy that allows Secrets Manager to use the key.
+
+   ```sh
+   aws kms put-key-policy --key-id key-id --policy-name default --policy file://key-policy.json
+   ```
+
+   Example `key-policy.json`:
+   ```json
+   {
+     "Version": "2012-10-17",
+     "Id": "key-default-1",
+     "Statement": [
+       {
+         "Sid": "Allow use of the key",
+         "Effect": "Allow",
+         "Principal": {
+           "Service": "secretsmanager.amazonaws.com"
+         },
+         "Action": [
+           "kms:Encrypt",
+           "kms:Decrypt",
+           "kms:GenerateDataKey",
+           "kms:DescribeKey"
+         ],
+         "Resource": "*"
+       }
+     ]
+   }
+   ```
+
+4. **Verify the Secret Configuration:**
+   Use the `describe-secret` command to verify that the secret is properly configured and associated with the correct KMS key.
+
+   ```sh
+   aws secretsmanager describe-secret --secret-id MySecretName
+   ```
+
+By following these steps, you ensure that Secrets Manager is in use with KMS, thereby enhancing the security of your secrets in AWS.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that Secrets Manager is in use for managing secrets in AWS KMS using Python scripts, you can follow these steps:
+
+### 1. **Install AWS SDK for Python (Boto3)**
+First, ensure you have Boto3 installed. If not, you can install it using pip:
+```bash
+pip install boto3
+```
+
+### 2. **Create a Secret in AWS Secrets Manager**
+Use the following Python script to create a secret in AWS Secrets Manager:
+
+```python
+import boto3
+import json
+
+# Initialize a session using Amazon Secrets Manager
+client = boto3.client('secretsmanager')
+
+# Create a secret
+response = client.create_secret(
+    Name='MySecretName',
+    Description='This is a secret for KMS',
+    SecretString=json.dumps({'username': 'myuser', 'password': 'mypassword'})
+)
+
+print(f"Secret created with ARN: {response['ARN']}")
+```
+
+### 3. **Retrieve the Secret from AWS Secrets Manager**
+Use the following Python script to retrieve the secret:
+
+```python
+import boto3
+import json
+
+# Initialize a session using Amazon Secrets Manager
+client = boto3.client('secretsmanager')
+
+# Retrieve the secret
+response = client.get_secret_value(
+    SecretId='MySecretName'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>